helm/thingsboard/templates/mqtt-transport.yaml

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright © 2016-2020 The Thingsboard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+apiVersion: apps/v1
+kind: {{ .Values.mqtt.kind }}
+metadata:
+  name: {{ .Release.Name }}-mqtt-transport
+  labels:
+    {{- include "thingsboard.labels" . | nindent 4 }}
+spec:
+{{- if not .Values.mqtt.autoscaling.enabled }}
+  replicas: {{ .Values.mqtt.replicaCount }}
+{{- end }}
+{{- if eq .Values.mqtt.kind "StatefulSet" }}
```

Comment:
why is it even an option for this to be a StatefulSet? Upstream docs suggests that? Yhis is a fully stateless microservice, it makes 0 sense to bind it to a k8s node...

Reply:
We usually deploy as Deployment , but upstream uses Statefulset:
https://github.com/thingsboard/thingsboard-ce-k8s/blob/master/minikube/thingsboard.yml#L61
I've given the option and set the default as Statefulset so it's more compilant with upstream but giving us the option to change.
I would leave as this